web/src/stream_edit.js

```diff
@@ -320,14 +320,14 @@ function stream_notification_reset(e) {
 
     stream_settings_api.bulk_set_stream_property(
         data,
-        $(e.target).closest(".subsection-parent").find(".alert-notification"),
+        $(this).closest(".subsection-parent").find(".alert-notification"),
     );
 }
 
-function stream_setting_changed(e) {
-    const sub = get_sub_for_target(e.target);
-    const $status_element = $(e.target).closest(".subsection-parent").find(".alert-notification");
-    const setting = e.target.name;
+function stream_setting_changed() {
```

Comment:
This helper and the one above it would probably read considerably better if they were passed element to interact with by the parent, and the parent was the only thing that interacted with `this`.

Reply:
Yeah that makes sense. Because arrow notation removes access to `this`, I ended up doing this:

```diff
     $("#channels_overlay_container").on(
         "click",
         ".subsection-parent .reset-stream-notifications-button",
-        stream_notification_reset,
+        function on_click() {
+            stream_notification_reset(this);
+        },
     );
 
     $("#channels_overlay_container").on(
         "change",
         ".sub_setting_checkbox .sub_setting_control",
-        stream_setting_changed,
+        function on_change() {
+            stream_setting_changed(this);
+        },
     );
```